gusto/linear_solvers.py

```diff
@@ -238,27 +231,23 @@ class IncompressibleSolver(TimesteppingSolver):
     :arg params: Solver parameters.
     """
 
-    def __init__(self, state, L, params=None):
-
-        self.state = state
-
-        if params is None:
-            self.params = {'ksp_type': 'gmres',
-                           'pc_type': 'fieldsplit',
-                           'pc_fieldsplit_type': 'additive',
-                           'fieldsplit_0_pc_type': 'lu',
-                           'fieldsplit_1_pc_type': 'lu',
-                           'fieldsplit_0_pc_factor_mat_solver_package': 'mumps',
-                           'fieldsplit_1_pc_factor_mat_solver_package': 'mumps',
-                           'fieldsplit_0_ksp_type': 'preonly',
-                           'fieldsplit_1_ksp_type': 'preonly'}
-        else:
-            self.params = params
+    default_solver_parameters = {
+        'ksp_type': 'gmres',
+        'pc_type': 'fieldsplit',
+        'pc_fieldsplit_type': 'additive',
+        'fieldsplit_0_pc_type': 'lu',
+        'fieldsplit_1_pc_type': 'lu',
+        'fieldsplit_0_pc_factor_mat_solver_package': 'mumps',
+        'fieldsplit_1_pc_factor_mat_solver_package': 'mumps',
+        'fieldsplit_0_ksp_type': 'preonly',
+        'fieldsplit_1_ksp_type': 'preonly'
+    }
```

Comment:
This is also something @wence- may have an opinion on. Should we be pushing to use the nested-dict syntax for solver options? For example, the options here can be written as
```(python)
default_solver_parameters = {'ksp_type': 'gmres',
                             'pc_type': 'fieldsplit',
                             'pc_fieldsplit_type': 'additive',
                             'fieldsplit_0': {'ksp_type': 'preonly',
                                              'pc_type' : 'lu',
                                              'pc_factor_mat_solver_package': 'mumps'},
                             'fieldsplit_1': {'ksp_type': 'preonly',
                                              'pc_type' : 'lu',
                                              'pc_factor_mat_solver_package': 'mumps'}}
```
Every sub-ksp (associated with `fieldsplit_x`) takes another dict of parameters. This, in my opinion, is a nice way to write out solver options as it seems a bit more readable. Not crucial, just a suggestion.